e2e/registration_test.go

```diff
@@ -197,6 +200,54 @@ service_registration {
 	assert.False(t, registered)
 }
 
+// TestE2E_ServiceRegistration_InitError tests that if the initial run
+// of all tasks fails, then CTS is not registered.
+func TestE2E_ServiceRegistration_InitError(t *testing.T) {
+	setParallelism(t)
+	srv := newTestConsulServer(t)
+	defer srv.Stop()
+
+	tempDir := fmt.Sprintf("%s%s", tempDirPrefix, "registration_init_failure")
+	cleanup := testutils.MakeTempDir(t, tempDir)
+	defer cleanup()
+
+	// Use a nonexistent module
```

Comment:
The module exists, it's just made to fail right?

Reply:
Oh, whoops, that was an old comment that I forgot about! I started with a nonexistent module, but realized that fails at the init stage and not at the plan/apply stage.